### PR TITLE
CosmWasm v1

### PR DESCRIPTION
**Pull request by assafmo**
Tasks: https://github.com/enigmampc/SecretNetwork/issues/770#issuecomment-905278597

### CODE REVIEW

File: cosmwasm/packages/wasmi-runtime/src/wasm/runtime/contract.rs [hunk 1]
```diff
@@ -73,24 +78,24 @@ impl ContractInstance {
...
-        if (&*module).export_by_name("cosmwasm_vm_version_3").is_some() {
-            cosmwasm_api_version = CosmWasmApiVersion::V010;
-        } else
-        /* if (&*module).export_by_name("interface_version_7").is_some() */
-        {
-            cosmwasm_api_version = CosmWasmApiVersion::V016;
-        }
+        if module.export_by_name(api_marker::V0_10).is_some() {
+            cosmwasm_api_version = CosmWasmApiVersion::V010
+        } else if module.export_by_name(api_marker::V0_16).is_some() {
```
[the remark this answers is not in the document]
Comment (assafmo) [reply cosmwasm/packages/wasmi-runtime/src/wasm/runtime/contract.rs:94]: Oh right, missed that, awesome! :100:

File: cosmwasm/packages/wasmi-runtime/src/wasm/runtime/contract.rs [hunk 2]
```diff
@@ -628,6 +628,12 @@ impl WasmiApi for ContractInstance {
             Ok(x) => x,
         };
 
+        if human_addr_str.is_empty() {
+            return Ok(Some(RuntimeValue::I32(
+                self.write_to_memory(b"input is empty")? as i32,
+            )));
+        }
+
```
Comment (assafmo) [inline cosmwasm/packages/wasmi-runtime/src/wasm/runtime/contract.rs:636]: `addr_validate` is a new API therefore this check is not necessary, please remove this.

Comment (reuvenpo) [reply cosmwasm/packages/wasmi-runtime/src/wasm/runtime/contract.rs:636]: It is necessary though:
https://github.com/CosmWasm/cosmwasm/blob/main/packages/vm/src/imports.rs#L129

Comment (assafmo) [reply cosmwasm/packages/wasmi-runtime/src/wasm/runtime/contract.rs:636]: We never looked at these, so there's no reason to start now.

**Pull request closed without merging**